### PR TITLE
Welcome Tour: Refactor and extract core functionality to /packages

### PR DESCRIPTION
**Pull request by chriskmnds**
## What this is about

This PR is Part 1 (alpha) of extracting the tour component into a publishable package for NPM.

Extracts the core functionality of the welcome tour and wraps it into a standalone, configurable package under `/packages` for general reuse. Bare-bones, nothing really styled. This would be equivalent to what we'd get with a Shepherd or other tour package, only more React-aware.

- Offers a simple API via configuration for enabling/disabling various features
- By default, the accessibility functionalities are provided: minimize/close on Esc, left-right key navigation

Check the [REAME.md](https://github.com/Automattic/wp-calypso/tree/add/welcome-tour-extract/packages/packaged-tour) for the more detailed overview. 

## Next up / on the horizon

We plan on extracting the rest of the welcome tour into a component for reuse also. That would include all the current setup - styling, controls, skeleton, minimized view, etc. To be decided whether the related APIs & data stores are also to be extracted and how.

## Testing

1. Ensure that the current welcome tour works as expected. Nothing should change functionality or design-wise.
    - `yarn install; yarn dev --sync` from `/apps/editing-tookit`
    - go to the editor, sidebar, and open the welcome guide
    - adding `&welcome-tour-next` to the URL should show the guided-tour with spotlight effects
2. Check the storybook examples demo ok: `yarn run packaged-tour:storybook:start`

## Next

Some follow-up issues for improving various parts will follow in the respective board.

### for shipping:

- [x] Improve README
- [x] Add storybook
- [x] A name (`packaged-tour` is a placeholder) 😎 
- [x] More testing, debugging, cleanup.

Screenshot 2021-11-24 at 9 15 23 AM

## Media

**Storybook:**

Kapture 2021-11-23 at 16 25 22

no flag | `&welcome-tour-next`
-- | --
Screenshot 2021-11-23 at 4 26 30 PM | Screenshot 2021-11-23 at 4 25 58 PM

### CODE REVIEW

File: apps/editing-toolkit/package.json [hunk 1]
```diff
@@ -57,6 +57,7 @@
 		"@automattic/launch": "^1.0.0",
 		"@automattic/newspack-blocks": "^1.37.1",
 		"@automattic/onboarding": "^1.0.0",
+		"@automattic/packaged-tour": "workspace:^",
```
Comment (chriskmnds) [inline apps/editing-toolkit/package.json]: this.. ❓ 🦆

Comment (noahtallen) [reply apps/editing-toolkit/package.json]: I think that should work!

**Comment by chriskmnds** [comment]
> Screenshot 2021-11-24 at 9 15 23 AM

Screenshot 2021-11-25 at 2 02 05 PM

Oh cmon! 😆

**Comment by chriskmnds** [comment]
🚢 🚢 🚢 🚢 🚢

**Pull request merged**